source/main/dqm4hep-mysql-set-parameter.cc

```diff
@@ -0,0 +1,157 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DQM4HEP.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Remi Ete
+ * @copyright CNRS , IPNL
+ *
+ * Create a quality test template file a ROOT file.
+ * Parses the different directories and populate the
+ * qtest file with found objects.
+ */
+
+// -- dqm4hep headers
+#include <dqm4hep/StatusCodes.h>
+#include <dqm4hep/Internal.h>
+#include <dqm4hep/DBInterface.h>
+
+// -- tclap headers
+#include "tclap/CmdLine.h"
+#include "tclap/Arg.h"
+
+using namespace std;
+using namespace dqm4hep::core;
+
+
+int main(int argc, char* argv[])
+{
+  std::string cmdLineFooter = "Please report bug to <dqm4hep@gmail.com>";
+  std::unique_ptr<TCLAP::CmdLine> pCommandLine(new TCLAP::CmdLine(cmdLineFooter, ' ', DQMCore_VERSION_STR));
+  
+  TCLAP::ValueArg<std::string> dbHostArg(
+      "k"
+      , "host"
+      , "The database host"
+      , false
+      , "localhost"
+      , "string");
+  pCommandLine->add(dbHostArg);
+  
+  TCLAP::ValueArg<std::string> dbUserArg(
+      "u"
+      , "user"
+      , "The database user"
+      , false
+      , "DQM4HEP"
+      , "string");
+  pCommandLine->add(dbUserArg);
+  
+  TCLAP::ValueArg<std::string> dbDbArg(
+      "d"
+      , "database"
+      , "The database name"
+      , false
+      , "DQM4HEP"
+      , "string");
+  pCommandLine->add(dbDbArg);
+  
+  TCLAP::ValueArg<std::string> dbPasswordArg(
+      "p"
+      , "password"
+      , "The database password"
+      , false
+      , ""
+      , "string");
+  pCommandLine->add(dbPasswordArg);
+  
+  TCLAP::ValueArg<std::string> dbTableArg(
+      "t"
+      , "table"
+      , "The database table to dump"
+      , true
+      , ""
+      , "string");
+  pCommandLine->add(dbTableArg);
+  
+  TCLAP::ValueArg<std::string> parameterNameArg(
+      "n"
+      , "parameter-name"
+      , "The parameter name"
+      , true
+      , ""
+      , "string");
+  pCommandLine->add(parameterNameArg);
+  
+  TCLAP::ValueArg<std::string> parameterValueArg(
+      "v"
+      , "parameter-value"
+      , "The parameter value"
+      , true
+      , ""
+      , "string");
+  pCommandLine->add(parameterValueArg);
+
+  // parse command line
+  pCommandLine->parse(argc, argv);
+  
+  std::string password = dbPasswordArg.getValue();
+  std::string parameter = parameterNameArg.getValue();
+  std::string value = parameterValueArg.getValue();
+  StringMap parameters;
+  parameters[parameter] = value;
+  
+  if(parameter.size() > 256)
+  {
+    dqm_error( "Invalid parameter size, expected <= 256" );
+    return 1;
+  }
+  
+  if(password.empty())
+  {
+    char *promptPassword = getpass("Password : ");
+    
+    if(!promptPassword)
+    {
+      dqm_error( "No password specified !" );
+      return 1;
+    }
+    
+    password = promptPassword;
+  }
```

Comment:
So not possible to have an empty password for the db user?

Reply:
Yes exactly ! Every users need a password in this scheme. Nevertheless, the user "DQM4HEP" is always defined without password and SELECT permission only